pkg/providers/v1/instances_v2.go

```diff
@@ -45,6 +47,12 @@ func (c *Cloud) getProviderID(ctx context.Context, node *v1.Node) (string, error
 // InstanceExists returns true if the instance for the given node exists according to the cloud provider.
 // Use the node.name or node.spec.providerID field to find the node in the cloud provider.
 func (c *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
+	if time.Since(node.CreationTimestamp.Time) < c.nodeEventualConsistencyGracePeriod {
+		// recently-launched EC2 instances may not appear in `ec2:DescribeInstances`
+		// we return an error to cause the cloud-node-lifecycle-controller to ignore this node
```

Comment:
Is there any downside to just forcing it to fail? Could increase the error logs. Can we just move this check down into code so that we only return an error if it's within the grace period AND it doesn't have the eventually consistent fields set? That would drastically reduce the noise.

Reply:
Yep had the same thought. We have to handle `InvalidInstanceId.NotFound` differently in the `Instances` v1/v2 impls, since we only have the context to apply this grace period in v2